python/python/res/server/simulation_context.py

```diff
@@ -122,3 +122,46 @@ def get_run_context(self):
 
     def stop(self):
         self._queue_manager.stop_queue( )
+
+
+    def job_status(self, iens, num_retry = 10):
+        """Will return the status of the job.
+
+        The status will be returned as a tuple where the first element is the
+        global status of the whole job, and the second element is a more
+        detailed status of the different elements in the forward model:
+
+           global_status, detailed_status = ctx.job_status(4)
+
+        The global_status will be obtained from the queue system and will
+        concern the status of the job as a whole. The detailed status is
+        obtained by reading a file from the filesystem, that file is typically
+        created by another process running on another machine, and reading
+        might fail due to NFS issues, simultanoues write and so on. If loading
+        valid json fails the function will sleep 0.10 seconds and retry -
+        eventually giving up and returning None.
+
+        The details object will look like this:
+
+           details.start_time
+           details.end_time
+           details.run_id
+
+           jobs: [ (job1.name, job1.start_time, job1.end_time, job1.status, job1.error_msg),
+                   (job2.name, job2.start_time, job2.end_time, job2.status, job2.error_msg),
+                   ....
+                   (jobN.name, jobN.start_time, jobN.end_time, jobN.status, jobN.error_msg) ]
+
+           The element in the jobs list can be access with details[].
+
+        """
+        if not iens in self._run_args:
+            return (None,None)
```

Comment:
Shouldn't this raise a `KeyError` or perhaps `ValueError`?

Reply:
Yes - I agree.